File: Array_store.java
import java.util.Scanner;

public class Array_store {
    private int[] arr = new int[10];

    public void acceptInput() {
        Scanner scanner = new Scanner(System.in);
        System.out.println("Enter 10 integers:");
        for (int i = 0; i < arr.length; i++) {
            arr[i] = scanner.nextInt();
        }
    }

    public void displayWithForLoop() {
        System.out.println("Array elements (using for loop):");
        for (int i = 0; i < arr.length; i++) {
            System.out.print(arr[i] + " ");
        }
        System.out.println();
    }

    public void displayWithWhileLoop() {
        System.out.println("Array elements (using while loop):");
        int i = 0;
        while (i < arr.length) {
            System.out.print(arr[i] + " ");
            i++;
        }
        System.out.println();
    }

    public void sortArray() {
        for (int i = 0; i < arr.length - 1; i++) {
            for (int j = 0; j < arr.length - i - 1; j++) {
                if (arr[j] > arr[j + 1]) {
                    // swap
                    int temp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = temp;
                }
            }
        }
        System.out.println("Array sorted.");
    }

    public int countOccurrences(int num) {
        int count = 0;
        for (int value : arr) {
            if (value == num) {
                count++;
            }
        }
        return count;
    }

    public void insertAtPosition(int num, int position) {
        if (position < 0 || position >= arr.length) {
            System.out.println("Invalid position!");
            return;
        }
        arr[position] = num;
        System.out.println("Number inserted at position " + position);
    }

    public int[] removeDuplicates() {
        int[] temp = new int[arr.length];
        int size = 0;

        for (int i = 0; i < arr.length; i++) {
            boolean found = false;
            for (int j = 0; j < size; j++) {
                if (arr[i] == temp[j]) {
                    found = true;
                    break;
                }
            }
            if (!found) {
                temp[size++] = arr[i];
            }
        }

        int[] result = new int[size];
        for (int i = 0; i < size; i++) {
            result[i] = temp[i];
        }
        return result;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        Array_store store = new Array_store();

        store.acceptInput();

        store.displayWithForLoop();
        store.displayWithWhileLoop();

        store.sortArray();
        store.displayWithForLoop();

        System.out.print("Enter a number to count its occurrences: ");
        int num = scanner.nextInt();
        int count = store.countOccurrences(num);
        System.out.println("Number " + num + " occurs " + count + " times.");

        System.out.print("Enter number to insert: ");
        int insertNum = scanner.nextInt();
        System.out.print("Enter position: ");
        int position = scanner.nextInt();
        store.insertAtPosition(insertNum, position);
        store.displayWithForLoop();

        int[] noDuplicates = store.removeDuplicates();
        System.out.println("Array without duplicates:");
        for (int val : noDuplicates) {
            System.out.print(val + " ");
        }
        System.out.println();
    }
}

File: array_store.py
class ArrayStore:
    def __init__(self):
        """Initializes an empty list to store integers."""
        self.data = []

    def accept_integers(self):
        """Accepts 10 integers from the user and stores them."""
        print("Please enter 10 integers, one at a time.")
        while len(self.data) < 10:
            try:
                num = int(input(f"Enter number {len(self.data) + 1}: "))
                self.data.append(num)
            except ValueError:
                print("Invalid input. Please enter an integer.")

    def display_with_for(self):
        """Displays all elements using a for loop."""
        print("\n--- Displaying with for loop ---")
        if not self.data:
            print("The list is empty.")
            return
        for item in self.data:
            print(item, end=" ")
        print()

    def display_with_while(self):
        """Displays all elements using a while loop."""
        print("\n--- Displaying with while loop ---")
        if not self.data:
            print("The list is empty.")
            return
        i = 0
        while i < len(self.data):
            print(self.data[i], end=" ")
            i += 1
        print()

    def sort_list(self):
        """Sorts the list in ascending order."""
        self.data.sort()
        print("\nList has been sorted in ascending order.")

    def count_occurrences(self, number):
        """Returns how many times a number appears in the list."""
        count = self.data.count(number)
        print(f"\nThe number {number} appears {count} time(s).")
        return count

    def insert_at_position(self, number, position):
        """Inserts a number at a specific position."""
        if 0 <= position <= len(self.data):
            self.data.insert(position, number)
            print(f"\nInserted {number} at position {position}.")
        else:
            print(f"Error: Position {position} is out of bounds.")

    def get_unique_elements(self):
        """Returns a new list containing only unique elements."""
        unique_list = list(set(self.data))
        print("\n--- Unique Elements ---")
        print(unique_list)
        return unique_list

if __name__ == "__main__":
    store = ArrayStore()

    store.accept_integers()

    store.display_with_for()
    store.display_with_while()

    store.sort_list()
    store.display_with_for()

    num_to_find = int(input("\nEnter a number to count its occurrences: "))
    store.count_occurrences(num_to_find)

    num_to_insert = int(input("\nEnter a number to insert: "))
    pos_to_insert = int(input(f"Enter the position to insert {num_to_insert}: "))
    store.insert_at_position(num_to_insert, pos_to_insert)
    store.display_with_for()

    store.get_unique_elements()